plugins/modules/purefa_smtp.py

```diff
@@ -72,61 +102,138 @@
 RETURN = r"""
 """
 
+HAS_PURESTORAGE = True
+try:
+    from pypureclient.flasharray import SmtpServer
+except ImportError:
+    HAS_PURESTORAGE = False
+
 from ansible.module_utils.basic import AnsibleModule
 from ansible_collections.purestorage.flasharray.plugins.module_utils.purefa import (
-    get_system,
+    get_array,
     purefa_argument_spec,
 )
 
 
 def delete_smtp(module, array):
     """Delete SMTP settings"""
     changed = True
+    api_version = array.get_rest_version()
     if not module.check_mode:
-        try:
-            array.set_smtp(
-                sender_domain="None", user_name="", password="", relay_host=""
+        res = array.patch_smtp_servers(
+            smtp=SmtpServer(
+                sender_domain="None",
+                user_name="",
+                password="",
+                relay_host="",
+                encryption_mode="",
+                sender_username="",
+                subject_prefix="",
+                body_prefix="",
+            )
+        )
+        if res.status_code != 200:
+            module.fail_json(
+                msg="Delete SMTP settigs failed. Error: {0}".foramt(
+                    res.errors[0].message
+                )
             )
-        except Exception:
-            module.fail_json(msg="Delete SMTP settigs failed")
     module.exit_json(changed=changed)
 
 
 def create_smtp(module, array):
     """Set SMTP settings"""
-    changed = changed_sender = changed_relay = changed_creds = False
-    current_smtp = array.get_smtp()
+    api_version = array.get_rest_version()
+    changed = False
+    # Currently only 1 SMTP server is configurable
+    current_smtp = list(array.get_smtp_servers().items)[0]
+    current_server = {
+        "sender_domain": current_smtp.sender_domain,
+        "relay_host": getattr(current_smtp, "relay_host", None),
+        "user_name": getattr(current_smtp, "user_name", None),
+        "encryption_mode": getattr(current_smtp, "encryption_mode", None),
+        "sender_username": getattr(current_smtp, "sender_username", None),
+        "subject_prefix": getattr(current_smtp, "subject_prefix", None),
+        "body_prefix": getattr(current_smtp, "body_prefix", None),
+    }
+    new_server = {
+        "sender_domain": current_smtp.sender_domain,
+        "relay_host": getattr(current_smtp, "relay_host"),
+        "user_name": getattr(current_smtp, "user_name", ""),
+        "encryption_mode": getattr(current_smtp, "encryption_mode", ""),
+        "sender_username": getattr(current_smtp, "sender_username", ""),
+        "subject_prefix": getattr(current_smtp, "subject_prefix", ""),
+        "body_prefix": getattr(current_smtp, "body_prefix", ""),
+    }
+
     if (
         module.params["sender_domain"]
-        and current_smtp["sender_domain"] != module.params["sender_domain"]
+        and current_server.sender_domain != module.params["sender_domain"]
```

Comment:
It should be current_server["sender_domain"], instead of current_server.sender_domain